### ISSUE TITLE
EOS method naming is inconsistent

### ISSUE THREAD

**Issue by inducer** [illinois-ceesd/mirgecom#254]
From #236:
```
    .. automethod:: get_density
    .. automethod:: get_species_molecular_weights
    .. automethod:: get_production_rates
    .. automethod:: get_species_source_terms
    .. automethod:: get_internal_energy
    .. automethod:: species_fractions
    .. automethod:: total_energy
    .. automethod:: gamma
    .. automethod:: gas_const
```
Some things have `get`, some others do not.

cc @MTCam

**Comment by matthiasdiener** [comment]
Maybe also declare them as `@property`?

**Comment by MTCam** [comment]
We discussed this previously, but long ago.

Then, we decided it was ok for some EOS functions to have names that are *not* verbs. `gamma`, `gas_const`, `total_energy` - because it makes the code easier to read at the "call sites" for these functions.   

For the mixture (pyrometheus) EOS, the `.. automethod: gamma` and similar are added because I wanted to override the documentation from the base class.   The `get_xxx` methods are not reflected in other non-mixture EOS, and specifically indicate those functions where we are retrieving a calculated quantity from the underlying `Pyrometheus` mechanism.

**Comment by inducer** [comment]
> Then, we decided it was ok for some EOS functions to have names that are _not_ verbs. `gamma`, `gas_const`, `total_energy` - because it makes the code easier to read at the "call sites" for these functions.

I agree so far.

> The `get_xxx` methods are not reflected in other non-mixture EOS, and specifically indicate those functions where we are retrieving a calculated quantity from the underlying `Pyrometheus` mechanism.

Does it matter, as far as the interface is concerned? Does the user need to know?

**Comment by MTCam** [comment]
> Does it matter, as far as the interface is concerned? Does the user need to know?

In many cases (informed) users want/need to know which routines are engaging the mechanism code to perform some sort of computation.  I would be OK with renaming some of these from `get_` -> `compute_`. It would also be OK to rename `get_species_molecular_weights -> species_molecular_weights`.

For the rest, I would suggest replacing with names like:

`compute_density`, `compute_internal_energy`, `compute_production_rates`, `compute_species_source_terms` 

We should bounce this off an informed user.  @anderson2981?

**Comment by anderson2981** [comment]
I agree with leaving in verbiage to let the user know when a more extensive computation is going to occur. This could avoid situations in the future where pyro is being called needlessly over and over again, instead of grabbing the data once and storing it. 

This is something that could potentially be caught in review and amended regardless, but for developers standing up new capabilities it would be useful I think. 

I actually prefer `compute_` over `get_` as it indicates a stronger computation load.

**Comment by inducer** [comment]
`compute_` for expensive things and attributes/methods without prefix  for cheap things sound good to me.

- [ ] This convention should be documented.
- [ ] Where do we draw the line for "expensive"? Any vector arithmetic at all? Evaluates a polynomial? Needs to iterate to do its job?

**Comment by MTCam** [comment]
> `compute_` for expensive things and attributes/methods without prefix for cheap things sound good to me.
> Where do we draw the line for "expensive"? Any vector arithmetic at all? Evaluates a polynomial? Needs to iterate to do its job?

I'm just not we should commit to a documented convention.  This approach has the potential to blow a lot of stuff up that already works as intended, and introduces more grayness and potential confusion than it solves, imo.

Take `pressure` and `temperature`, for a straightforward example.  For single component materials, these are trivial, but for mixtures both calls require a Newton solve.

**Comment by inducer** [comment]
>  This approach has the potential to blow a lot of stuff up that already works as intended

How so? (If you mean the renames, those are usually easy and low risk in Python, at least IMO.

> Take `pressure` and `temperature`, for a straightforward example. For single component materials, these are trivial, but for mixtures both calls require a Newton solve.

That, to me, would speak against trying to encode computational effort into a name.

**Comment by MTCam** [comment]
> > This approach has the potential to blow a lot of stuff up that already works as intended
> 
> How so? (If you mean the renames, those are usually easy and low risk in Python, at least IMO.
> 

I mean because it takes something already works as it is intended to work, with a uniform interface regardless of number of mixture components, and introduces a renaming scheme and convention that doesn't hold in most cases - just to address a cosmetics-type issue. 

> > Take `pressure` and `temperature`, for a straightforward example. For single component materials, these are trivial, but for mixtures both calls require a Newton solve.
> 
> That, to me, would speak against trying to encode computational effort into a name.

I do not disagree (but @anderson2981 might).  My suggestion was to replace `get_` -> `compute_` for routines that reach into the pyrometheus back-end simply because `get_` naming appears to be unliked. Even `get_` would not have been there if it weren't an implied directive to use a verb for class method names.

Maybe the best way would be to remove the distinction altogether and just forgo the notion that we need verbs for EOS function names, period?

**Comment by inducer** [comment]
> Maybe the best way would be to remove the distinction altogether and just forgo the notion that we need verbs for EOS function names, period?

I'm OK with that.

> just to address a cosmetics-type issue.

Sorry. Not meaning to waste time with bikeshedding.

**Comment by MTCam** [comment]
> > Maybe the best way would be to remove the distinction altogether and just forgo the notion that we need verbs for EOS function names, period?
> 
> I'm OK with that.

Perhaps this is what we should do, but let's see if we take any fire from @anderson2981 first?

> > just to address a cosmetics-type issue.
> 
> Sorry. Not meaning to waste time with bikeshedding.

I agree 100% that consistency is important, and I am not great at catching when things aren't. The point I failed to make earlier is that these particular routines are not present in non-mixture EOS - so there's nothing to be consistent with - which makes it (imo) a cosmetics-type issue.

**Comment by MTCam** [comment]
At least related if not duplicate: #485
